modules/apps/27-interchain-accounts/module.go

```diff
@@ -57,34 +67,34 @@ func (AppModule) IsOnePerModuleType() {}
 // IsAppModule implements the appmodule.AppModule interface.
 func (AppModule) IsAppModule() {}
 
-// RegisterLegacyAminoCodec implements AppModuleBasic.
-func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {}
+// RegisterLegacyAminoCodec implements AppModule.
+func (AppModule) RegisterLegacyAminoCodec(cdc coreregistry.AminoRegistrar) {}
```

Comment:
is this solely to fulfil the `AppModuleBasic` interface we currently assert on line 34? Can't we just fulfil the `HasGRPCGateway` https://github.com/cosmos/cosmos-sdk/blob/5799c15b90998b1a1474c50695f0d5063af6bd38/types/module/module.go#L52 interface?

Reply:
ah yes, that's correct. There is no implementation here. 

From what I understand we can remove the assertion on `AppModuleBasic` and only use `HasGRPCGateway`. 

I think we should comb through all of the modules to ensure we have the minimum amount of interface assertions needed. 

What do you think about creating an issue for this and doing it post-merge?

Reply:
sounds good to me, I see we're doing diff assertions in some of the modules atm so an issue to comb through and axe/normalize all seems like a plan

Reply:
#7701